charts/sophora-server/templates/role.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.serverModeLabeler.enabledOnClusterServers .Values.serverModeLabeler.createServiceAccount }}
+{{- if and .Values.serverModeLabeler.enabled .Values.serviceAccount.create }}
```

Comment:
```suggestion
{{- if .Values.serviceAccount.create -}}
```

Reply:
What about renaming this `Role` and `RoleBinding` to 
```
{{ include "sophora-server.fullname" . }}-serverModeLabeler
```
and let the conditions as is?

With more requirements emerging we could just add another `Role` and `RoleBinding` so we're sticking to POLP based on what the SA should really be allowed to do.

Reply:
Based on our chat discussion, I believe this could be one possible approach. However, we would need to implement a helper function for the names to prevent them from being too long when `sophora-server.fullname` exceeds 63 characters minus the length of the suffix for the role. 

Would you implement this?

Reply:
I'm not sure what to do in this case. Just cutting the name after 63 characters feels like a hacky solution.
Maybe we just throw a validation error or let Kubernetes take care of it on apply?

Reply:
Validating it via Helm is ok for new installations. Existing installations which may run into this problem would then need to be re-installed completely.

Curious how other charts do this... Maybe we could trim it in a way that the prefix is trimmed, not the suffix. This would ensure that Roles don't accidentally get the same name due to the trimming.

Reply:
We could do sth. like this
```
  name: {{ include "sophora-server.fullname" . | trunc (sub 63 (len "serverModeLabeler") | int) }}-serverModeLabeler
```

With a pretty long value for `nameOverride` like `sophora-server-this-is-a-very-long-nameOverride` we get this
```
-> % helm template -f test-values.yaml test . | yq 'select(.kind == "Role")'
# Source: sophora-server/templates/role.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: test-sophora-server-this-is-a-very-long-nameOv-serverModeLabeler
```

Reply:
Yep, sounds good. Should be `len "-serverModeLabeler"` though. :) 

And we could extract that into a helper function so we can call this with various suffixes.